hdx_hapi/endpoints/get_population.py

```diff
@@ -31,56 +30,35 @@
 
 
 @router.get(
-    '/api/themes/population',
+    '/api/population-social/population',
     response_model=HapiGenericResponse[PopulationResponse],
     summary='Get baseline population data',
     include_in_schema=False,
 )
 @router.get(
-    '/api/v1/themes/population',
+    '/api/v1/population-social/population',
     response_model=HapiGenericResponse[PopulationResponse],
     summary='Get baseline population data',
 )
 async def get_populations(
     common_parameters: Annotated[CommonEndpointParams, Depends(common_endpoint_parameters)],
+    ref_period_parameters: Annotated[ReferencePeriodParameters, Depends(reference_period_parameters)],
     db: AsyncSession = Depends(get_db),
-    gender_code: Annotated[str, Query(max_length=1, description='Gender code')] = None,
-    age_range_code: Annotated[str, Query(max_length=32, description='Age range code')] = None,
+    gender: Annotated[str, Query(max_length=16, description='Gender')] = None,
```

Comment:
Gender is now an Enum in `enums.py` (from the schema library) . I think we should _type_ it as  such, instead of a `str`, as this will be reflected in the OpenAPI specs. The same goes for the other functions where gender appears and also in the Pydantic model.

Reply:
I wondered about this - I had problems with doing the gender query using the insensitive case matcher because of a typing mismatch.

I'll fix it.

Reply:
I fixed this in 8d9d0aaa684a77a94c933f8d1394b0ec97dfae26 but just for the population, there are gender fields in refugees and humanitarian_needs.